components/nexus-core/src/main/java/org/sonatype/nexus/internal/blobstore/BlobStoreManagerImpl.java

```diff
@@ -81,39 +119,102 @@ protected void doStop() throws Exception {
 
   @Override
   @Guarded(by = STARTED)
+  public Iterable<BlobStore> browse() {
+    return ImmutableList.copyOf(stores.values());
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public BlobStore create(final BlobStoreConfiguration configuration) throws Exception {
+    checkNotNull(configuration);
+    log.debug("Creating blob-store: {} with content root: {}", configuration.getName(),
+        configuration.getPath());
+
+    store.create(configuration);
+
+    BlobStore blobStore = newBlobStore(configuration);
+    track(configuration.getName(), blobStore);
+    
+    blobStore.start();
+    //TODO - event publishing
+
+    return blobStore;
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public BlobStore update(final BlobStoreConfiguration configuration) throws Exception {
+    checkNotNull(configuration);
+    String name = checkNotNull(configuration.getName());
+    String path = checkNotNull(configuration.getPath());
+    
+    log.debug("Updating BlobStore: {}, {}", name, path);
+    
+    //TODO - changing the path will leave behind previous content, but is that sensible?
```

Comment:
It does seem strange. Why would someone want to change the path and only the path?

Alternatively, can we drop `update` and make people use a combination of `delete` and `create` instead?